## Informe de Calidad de los datos "reviews_Florida" proveniente de Google Maps:

In [1]:
#Librerias a utilizar
import pandas as pd
import json

 - Leer los archivo JSON en un DataFrame de Pandas

In [2]:
reviews_01 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/1.json', lines=True)

In [3]:
reviews_02 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/2.json', lines=True)

In [4]:
reviews_03 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/3.json', lines=True)

In [5]:
reviews_04 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/4.json', lines=True)

In [6]:
reviews_05 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/5.json', lines=True)

In [7]:
reviews_06 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/6.json', lines=True)

In [8]:
reviews_07 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/7.json', lines=True)

In [9]:
reviews_08 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/8.json', lines=True)

In [10]:
reviews_09 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/9.json', lines=True)

In [11]:
reviews_10 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/10.json', lines=True)

In [12]:
reviews_11 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/11.json', lines=True)

In [13]:
reviews_12 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/12.json', lines=True)

In [14]:
reviews_13 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/13.json', lines=True)

In [15]:
reviews_14 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/14.json', lines=True)

In [16]:
reviews_15 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/15.json', lines=True)

In [17]:
reviews_16 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/16.json', lines=True)

In [18]:
reviews_17 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/17.json', lines=True)

In [19]:
reviews_18 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/18.json', lines=True)

In [20]:
reviews_19 = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Google Maps/archivos reviws/19.json', lines=True)

- Tamaño de cada archivo

In [21]:
print(reviews_01.shape)
print(reviews_02.shape)
print(reviews_04.shape)
print(reviews_05.shape)
print(reviews_06.shape)
print(reviews_07.shape)
print(reviews_08.shape)
print(reviews_09.shape)
print(reviews_10.shape)
print(reviews_11.shape)
print(reviews_12.shape)
print(reviews_13.shape)
print(reviews_14.shape)
print(reviews_15.shape)
print(reviews_16.shape)
print(reviews_17.shape)
print(reviews_18.shape)
print(reviews_19.shape)

(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)
(150000, 8)


- Concatenar los dataframes usando la función concat

In [22]:
reviews = pd.concat(
    [reviews_01, reviews_02, reviews_03, reviews_04, reviews_05, reviews_06, reviews_07, reviews_08, reviews_09, reviews_10, 
     reviews_11, reviews_12, reviews_13, reviews_14, reviews_15, reviews_16, reviews_17, reviews_18, reviews_19]
    )

In [23]:
# Vista
reviews.head()

user_id              name           time  rating  \
0  1.014719e+20   Julie A. Gerber  1628003250740       1   
1  1.154772e+20  Martin Sheffield  1595031217005       5   
2  1.018050e+20      Brian Truett  1522924253567       5   
3  1.063444e+20          Tina Sun  1467907819586       1   
4  1.008751e+20      James Haynes  1480683415081       5   

                                                text  pics  \
0  Update: Their “reply” to my review amounted to...  None   
1  He's a knowledgeable doctor but the way he run...  None   
2  Best doctor I've ever had, I never wait to be ...  None   
3  I was told he is a good doctor. I was trying t...  None   
4  Takes the time to actually get to know his pat...  None   

                                                resp  \
0  {'time': 1627042799532, 'text': 'Thank you for...   
1  {'time': 1582464056733, 'text': 'Thank you for...   
2                                               None   
3                                               None   
4                                               None   

                                 gmap_id  
0  0x8893863ea87bd5dd:0x9383ebf973e74abb  
1  0x8893863ea87bd5dd:0x9383ebf973e74abb  
2  0x8893863ea87bd5dd:0x9383ebf973e74abb  
3  0x8893863ea87bd5dd:0x9383ebf973e74abb  
4  0x8893863ea87bd5dd:0x9383ebf973e74abb

In [24]:
# Columnas
reviews. columns

Index(['user_id', 'name', 'time', 'rating', 'text', 'pics', 'resp', 'gmap_id'], dtype='object')

Etiquetas que contien los nombre de las columnas:

'user_id': representa el ID del usuario que ha realizado la reseña.

'name': representa el nombre del establecimiento que ha sido objeto de la reseña.

'time': representa el momento en el que se realizó la reseña.

'rating': representa la calificación otorgada por el usuario al establecimiento.

'text': representa el texto de la reseña realizada por el usuario.

'pics': representa cualquier imagen adjunta a la reseña.

'resp': representa la respuesta del establecimiento a la reseña.

'gmap_id': representa el ID del establecimiento en Google Maps.

In [25]:
# Total de datos foltrados por florida
reviews.shape

(2850000, 8)

El archivo concatenado contiene 2850000 filas y 8 columnas.

In [26]:
# Obtener información sobre los tipos de datos, valores faltantes 
reviews.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2850000 entries, 0 to 149999
Data columns (total 8 columns):
 #   Column   Dtype  
---  ------   -----  
 0   user_id  float64
 1   name     object 
 2   time     int64  
 3   rating   int64  
 4   text     object 
 5   pics     object 
 6   resp     object 
 7   gmap_id  object 
dtypes: float64(1), int64(2), object(5)
memory usage: 195.7+ MB


La descripción indica que se trata de un objeto DataFrame de Pandas con las siguientes características:

Int64Index: Tiene un índice de tipo Int64, lo que significa que cada fila está identificada por un número entero único. El índice va desde 0 hasta 149999 y tiene una longitud de 2850000.

Data columns: Tiene un total de 8 columnas, cada una de las cuales contiene datos de diferentes tipos:

user_id: float64
name: object 
time     int64  
rating   int64  
text     object 
pics     object 
resp     object 
gmap_id  object 

Para cada columna, se muestra el número de valores no nulos, indicando que algunas columnas tienen valores faltantes (NaN).

memory usage: Indica el tamaño de memoria utilizado por el DataFrame que es de aproximadamente 36.1 megabytes.

In [27]:
# Estadísticas básicas de los datos
reviews.describe()

user_id          time        rating
count  2.850000e+06  2.850000e+06  2.850000e+06
mean   1.093032e+20  1.555140e+12  4.313658e+00
std    5.276713e+18  4.360793e+10  1.170064e+00
min    1.000000e+20  1.041379e+12  1.000000e+00
25%    1.047751e+20  1.527462e+12  4.000000e+00
50%    1.092897e+20  1.557678e+12  5.000000e+00
75%    1.138648e+20  1.585605e+12  5.000000e+00
max    1.184467e+20  1.631205e+12  5.000000e+00

In [30]:
# Datos nulos de cada columna 
reviews.isnull().sum()

user_id          0
name             0
time             0
rating           0
text       1079510
pics       2745810
resp       2394601
gmap_id          0
dtype: int64

Número de valores nulos para cada columna en el DataFrame.

text: es una columna que contiene el texto de la reseña. Hay un total de 1,079,510 valores faltantes en esta columna, lo que indica que algunos usuarios no han dejado ningún comentario en su reseña. Se considerara eliminar la columna.

pics: es una columna que indica si se incluyó alguna foto en la reseña. Hay un total de 2,745,810 valores faltantes en esta columna, lo que indica que la mayoría de las reseñas no incluyen fotos. Se considerara eliminar la columna.

resp: es una columna que indica si el negocio o alguien relacionado con él ha respondido a la reseña del usuario. Hay un total de 2,394,601 valores faltantes en esta columna, lo que indica que la mayoría de las reseñas no han recibido ninguna respuesta del negocio. Se considerara eliminar la columna.

In [ ]:
# Eliminar {} en la columna "resp"
reviews['resp'] = reviews['resp'].apply(lambda x: json.dumps(x))
reviews['resp'] = reviews['resp'].apply(lambda x: x.replace('{', '').replace('}', ''))

In [33]:
# Reemplazar las listas por cadena de caracteres
reviews['user_id'] = reviews['user_id'].apply(lambda x: str(x).strip('[]'))
reviews['name'] = reviews['name'].apply(lambda x: str(x).strip('[]'))
reviews['time'] = reviews['time'].apply(lambda x: str(x).strip('[]'))
reviews['rating'] = reviews['rating'].apply(lambda x: str(x).strip('[]'))
reviews['text'] = reviews['text'].apply(lambda x: str(x).strip('[]'))
reviews['pics'] = reviews['pics'].apply(lambda x: str(x).strip('[]'))
reviews['resp'] = reviews['resp'].apply(lambda x: str(x).strip('[]'))
reviews['gmap_id'] = reviews['gmap_id'].apply(lambda x: str(x).strip('[]'))

In [34]:
# Encunetro duplicados
reviews.duplicated().sum()

119396

Se encuentran 119396 duplicados, que despues de ser analizados en profundidad se decidirá si elminarlos o fucionarlos.